week4/src/component/common/Button/Button.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import styled, { css } from 'styled-components';
+import { Theme } from '../../../styles/theme';
+
+interface ButtonProps {
+  text: string;
+  onClick?: () => void;
+  hoverColor?: string;
+  disabled?: boolean;
+}
+
+const StyledButton = styled.button<{ hoverColor?: string }>`
+  background-color: ${Theme.color.buttonColor};
+  padding: 0.5rem 1rem;
+  border: none;
+  cursor: pointer;
+  font-size: 1rem;
+  width: 25rem;
+  height: 3rem;
+  transition: background-color 0.3s ease, color 0.3s ease;
+
+  &:hover {
+    ${({ hoverColor }) =>
+      hoverColor
+        ? css`
+            background-color: ${hoverColor};
+          `
+        : css`
+            background-color: #0056b3;
+          `}
+  }
+
+  &:disabled {
+    background-color: #dcdcdc;
+    cursor: not-allowed;
+    &:hover {
+      background-color: #dcdcdc;
+    }
+  }
+`;
+
+const Button: React.FC<ButtonProps> = ({
+  text,
+  onClick,
+  hoverColor,
+  disabled,
+}) => {
```

Comment:
`React.FC`를 사용하셨는데, 이 문법은 React 18 이상에서는 사용하지 않는 문법입니다!

사용하지 않는 이유는 제네릭을 지원하지 않기도 하고, children이 기본적으로 들어가있기 때문인데요,
저희가 타입스크립트를 사용하는 이유는 타입을 정확히 지정해줘서 코드의 안정성을 향상시키기 위함인데요, 
`FC`를 사용하면 컴포넌트에 children일 있을 수 있다는 것을 가정해서 children을 타입 지정하지 않고 전달이 가능하기 때문에 오히려 안정성을 떨어뜨립니다!! 

또, 현재 이 코드에서는 children을 사용하지 않고 있기 때문에 FC를 사용하지 않는 것이 더 적합합니다!
```suggestion

const Button = ({
  text,
  onClick,
  hoverColor,
  disabled,
}: ButtonProps) => {
```
이렇게 작성해서 더 간결하고 명확하게 타입을 정의해보는 것은 어떨까요?

함께 읽어보면 좋을 것 같아서 아티클도 첨부합니다!!
https://f-lab.kr/insight/react-fc-type-usage-20241102